File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        // Brute Force Approach
        int longestSS=0; // initialised to 0
        for(int i=0;i<s.length();i++){
            Set<Character> set=new HashSet<>();
            for(int j=i;j<s.length();j++){
                if(set.contains(s.charAt(j))){
                    break;
                }
                set.add(s.charAt(j));
            }
            longestSS=Math.max(longestSS,set.size());
        }
        return longestSS;
    }
}

File: Data Structures & Algorithms/house-robber/submission-2.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        return self.dp(nums, len(nums) -1, {})

    def dp(self, nums: List[int], n: int, memo: dict):
        if n==0:
            return nums[0]
        
        if n==1:
            return max(nums[0], nums[1])
        
        if n in memo:
            return memo[n]
            
        memo[n]=max(self.dp(nums, n-1, memo), nums[n] + self.dp(nums, n-2, memo))

        return memo[n]